packages/firestore/src/local/simple_db.ts

```diff
@@ -181,6 +191,25 @@ export class SimpleDb {
     return txn.store<KeyType, ValueType>(store);
   }
 
+  /** Parse User Agent to determine iOS version. Returns -1 if not found. */
```

Comment:
Did you write these regex yourself or did you borrow them from a trustworthy source? 😄 If the former, I'll review them more closely...  If the latter, perhaps add a comment declaring where they came from.

Reply:
I combined them from several sources. 

iOS: [source](https://stackoverflow.com/questions/48009956/php-get-exact-ios-version-from-user-agent?rq=1): Original one was found in php, but javacsript does not support `\k` in regex queries, so I ran multiple manually to get the regex code right. Tested on iOS simulator with iphone + ipad.

Android: Looked at several android user agents, applied the same regex as iOS but with `.` instead of `_`.